pkg/operator/controller/garden/garden/components.go

```diff
@@ -660,45 +660,65 @@ func (r *Reconciler) newKubeAPIServer(
 		}
 	}
 
-	if enableSeedAuthorizer {
+	if enableAdmissionControllerAuthorizers {
 		caSecret, found := secretsManager.Get(operatorv1alpha1.SecretNameCAGardener)
 		if !found {
 			return nil, fmt.Errorf("secret %q not found", operatorv1alpha1.SecretNameCAGardener)
 		}
 
-		kubeconfig, err := runtime.Encode(clientcmdlatest.Codec, kubernetesutils.NewKubeconfig(
-			"authorization-webhook",
-			clientcmdv1.Cluster{
-				Server:                   fmt.Sprintf("https://%s/webhooks/auth/seed", gardeneradmissioncontroller.ServiceName),
-				CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle],
-			},
-			clientcmdv1.AuthInfo{},
-		))
+		var (
+			newAuthorizationKubeconfig = func(name string) ([]byte, error) {
+				return runtime.Encode(clientcmdlatest.Codec, kubernetesutils.NewKubeconfig(
+					name+"-authorization-webhook",
+					clientcmdv1.Cluster{
+						Server:                   fmt.Sprintf("https://%s/webhooks/auth/"+name, gardeneradmissioncontroller.ServiceName),
+						CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle],
+					},
+					clientcmdv1.AuthInfo{},
+				))
+			}
+			newAuthorizationWebhook = func(name string, kubeconfig []byte, matchExpression string) kubeapiserver.AuthorizationWebhook {
+				return kubeapiserver.AuthorizationWebhook{
+					Name:       name + "-authorizer",
```

Comment:
Nit: I am slightly confused by the spacing of string concatenations. Sometimes whitespace is added, sometimes it is omitted, e.g. `name + "-authorizer"` vs. `name+"-authorization-webhook"` a couple lines above. Is there some logic behind this, which I do not get?